### PR TITLE
added CDN settings to reaction

### LINKED ISSUE #4315: No native CDN support for production bundle in Reaction.
## Issue Description
After creating the production bundle from reaction, if the user wants to serve the bundled `js` and `css` files, there is not native way to do it.

## Steps to Reproduce
1. Build the production bundle from reaction using `meteor build --directory "<build dir>"` 
1. `cd "<build dir>"/bundle/programs/server`
1. `npm install`
1. `cd "<build dir>"/bundle`
1. `PORT=4000 ROOT_URL=http://localhost.com node main.js`

Browse to the URL and see that the `js` and `css` files are served from the server. But if you want to upload it to CDN and serve the files from there, there is no way to do it.

### PR DESCRIPTION
**Pull request by Akarshit**
Resolves #4315   
Impact: **minor**  
Type: **feature**

## Issue
After creating the production bundle from reaction, if the user wants to serve the bundled js and css files, there is no native way to do it.

## Solution
Meteor provides a function to set the serving URL of the `js` and `css` files, so I am just passing the CDN's URL from setting file to Meteor. 

## Testing
1. Build the production bundle from reaction using `meteor build --directory "build dir"`
1. `cd "build dir"/bundle/programs/server`
1. `npm install`
1. `cd "build dir"/bundle`
1. `PORT=4000 ROOT_URL=http://localhost.com METEOR_SETTINGS="{ \"cdnPrefix\": \"cdnURL\" }" node main.js`
1. Observe from the Network tab in your browser that reaction tries to get the `js` and `css` files from the "cdnURL"

### CODE REVIEW

File: server/startup/index.js [hunk 1]
```diff
@@ -26,6 +26,11 @@ SimpleSchema.defineValidationErrorTransform((error) => {
 export default function startup() {
   const startTime = Date.now();
 
+  // for the user who wants to serve bundled js and css files from different URL
```
Comment (zenweasel) [inline server/startup/index.js]: Let's put this in it's own file so we aren't anything extra to this file

**Comment by Akarshit** [comment]
@zenweasel updated.

File: server/startup/cdn.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Meteor } from "meteor/meteor";
+/**
+ * Sets prefix for the user who wants to serve bundled js and css
+ * files from different URL
+ *
+ * @returns null
+ */
+export default function CDN() {
```
Comment (zenweasel) [inline server/startup/cdn.js]: Function naming should follow the conventions in the style guide. Specifically this should be a verb giving you some sense of what this function does and camel casing includes acronyms. e.g. `setupCdn` . And it should include a JSDoc

Comment (zenweasel) [reply server/startup/cdn.js]: @Akarshit What's happening with this?

File: server/startup/cdn.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Meteor } from "meteor/meteor";
+
+/**
+ * Sets prefix for the user who wants to serve bundled js and css
```
Comment (zenweasel) [inline server/startup/cdn.js]: This is not in the correct JSDoc format. Please look at the documentation for the correct format.

Also, javascript functions don't return `null` by default they return `undefined` which is not the same thing

**Comment by spencern** [comment]
@zenweasel @Akarshit need another review here and to pull in the updated .snyk file from `master`

**Comment by Akarshit** [comment]
@spencern I have merge `release-1.14.0` into this, what else should I do?

**Comment by zenweasel** [comment]
@Akarshit I need to take another look. Will do it on Monday

**Comment by zenweasel** [comment]
When I follow the testing instructions I get
```
in the root directory of your application.
/Users/brenthoover/meteor-builds/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:218
      throw error;
      ^

Error: Cannot find module '/imports/plugins/included/taxes-taxjar/server'
    at makeMissingError (packages/modules-runtime.js:231:12)
    at require (packages/modules-runtime.js:241:19)
    at plugins.js (server/plugins.js:1:3133)
    at fileEvaluate (packages/modules-runtime.js:339:7)
    at require (packages/modules-runtime.js:238:16)
    at /Users/brenthoover/meteor-builds/bundle/programs/server/app/app.js:106915:1
    at /Users/brenthoover/meteor-builds/bundle/programs/server/boot.js:411:36
    at Array.forEach (<anonymous>)
    at /Users/brenthoover/meteor-builds/bundle/programs/server/boot.js:220:19
    at /Users/brenthoover/meteor-builds/bundle/programs/server/boot.js:471:5
    at Function.run (/Users/brenthoover/meteor-builds/bundle/programs/server/profile.js:510:12)
    at /Users/brenthoover/meteor-builds/bundle/programs/server/boot.js:470:11
```

**Comment by zenweasel** [comment]
Ok, figured out that problems (have to rerun `reaction plugins load` to clean out imports). but when I bring the app up I get this:
```
http://localhost:4000/cdnURL/5c9381ddcbcae43f804286621e340b1b0d34f882.js?meteor_js_resource=true 404 (Not Found)
```
Note that I am copying and pasting your instructions from above verbatim.

**Comment by Akarshit** [comment]
@zenweasel That's weird, because I just tested the instructions again and they work.
Except that MONGO_URL also needs to be passed in the final command.

**Comment by zenweasel** [comment]
Yes, here is the exactly line I am pasting;
```
PORT=4000 MONGO_URL=mongodb://localhost:27017/meteor ROOT_URL=http://localhost.com METEOR_SETTINGS="{ \"cdnPrefix\": \"cdnURL\" }" node main.js
```
And I get:
```
http://localhost:4000/cdnURL/5c9381ddcbcae43f804286621e340b1b0d34f882.js?meteor_js_resource=true
```

**Comment by zenweasel** [comment]
I am expecting to get:

```
cdnURL/5c9381ddcbcae43f804286621e340b1b0d34f882.js?meteor_js_resource=true
```

**Comment by Akarshit** [comment]
@zenweasel This is the intended behaviour
so if you put `cdnPrefix: hello`, the files are served from `ROOT_URL/cdnPrefix`
and if you put `cdnPrefix: http://rc.com`, the files are served from `http://rc.com/`

**Comment by zenweasel** [comment]
Got it. Could have been clearer in the test instructions.

**Comment by zenweasel** [comment]
@Akarshit We have a conflict here now

**Comment by Akarshit** [comment]
@zenweasel 
Updated.

**Pull request merged**